### ISSUE TITLE
Cat overrides attr.ib validators.

### ISSUE THREAD

**Issue by asford** [xoeye/typecats#1]
First of all, thanks for open-sourcing this tools. It's a great idea and super useful.

In an initial test, I've found that the the non-empty validator added by `Cat` overrides any standard attrs-based validators declared on the class. This is a little surprising, so I believe it's unintended behavior.

I believe this is due to the logic in [`_hack_add_validator`](https://github.com/xoeye/typecats/blob/7398cdf84487d5ec4bfa9fa8acb917d4d343ffc0/typecats/attrs_shim.py#L123).

As a minimal repro:

```python
import attr
import typecats
import pytest

def test_typecat_validators():
    @attr.s
    class Foo:
        bar: str = attr.ib()

        @bar.validator
        def _is_bar(self, _, value):
            assert value == "bar"

    with pytest.raises(AssertionError):
        Foo("not bar")

    Foo("bar")

    @typecats.Cat
    class FooCat:
        bar: str = attr.ib()

        @bar.validator
        def _is_bar(self, _, value):
            assert value == "bar"

    with pytest.raises(AssertionError):
        FooCat("not bar")

    FooCat("bar")
```

**Comment by petergaultney** [comment]
haha, note the name - "hack".  ;)

I would think we could find a way to wrap an existing validator rather than overwrite them. You're welcome to take a stab at making a PR for this functionality. It may take me a week or two to get back to this.

**Comment by petergaultney** [comment]
Hi Alex,

I think I've fixed this with typecats 1.1.1. I added your test and a few others for good measure. Let me know if you have any more issues, and thanks for using the library!

**Comment by asford** [comment]
@petergaultney Awesome, thanks for tackling this so quickly.
